br/pkg/aws/ebs.go

```diff
@@ -281,6 +281,155 @@ func (e *EC2Session) DeleteSnapshots(snapIDMap map[string]string) {
 	log.Info("delete snapshot end", zap.Int("need-to-del", len(snapIDMap)), zap.Int32("deleted", deletedCnt.Load()))
 }
 
+// EnableDataFSR enables FSR for data volume snapshots
+func (e *EC2Session) EnableDataFSR(meta *config.EBSBasedBRMeta, targetAZ string) (map[string][]*string, error) {
+	snapshotsIDsMap := fetchTargetSnapshots(meta, targetAZ)
+
+	if len(snapshotsIDsMap) == 0 {
+		return snapshotsIDsMap, errors.Errorf("empty backup meta")
+	}
+
+	eg, _ := errgroup.WithContext(context.Background())
+
+	workerPool := utils.NewWorkerPool(e.concurrency, "enable snapshot FSR")
+
+	for availableZone := range snapshotsIDsMap {
+		workerPool.ApplyOnErrorGroup(eg, func() error {
+			log.Info("enable fsr for snapshots", zap.String("available zone", availableZone))
+			resp, err := e.ec2.EnableFastSnapshotRestores(&ec2.EnableFastSnapshotRestoresInput{
+				AvailabilityZones: []*string{&availableZone},
+				SourceSnapshotIds: snapshotsIDsMap[availableZone],
```

Comment:
it seems race here. `availableZone` is a loop variable which cannot be capture by another goroutine. need a copy of `availableZone` here.